app.py

```diff
@@ -1,27 +1,29 @@
 from pathlib import Path
 
 import gradio as gr
-from langchain import OpenAI, SQLDatabase, SQLDatabaseChain
+from langchain import SQLDatabase, SQLDatabaseChain
+from langchain.chat_models import ChatOpenAI
+from langchain.llms.fake import FakeListLLM
 
-# Define some example queries
-examples = [
-    ["How many players are there with a first name that begins with 'John'?"],
-    ["Is steph curry active?"],
-    ["List all players with a last name of Jordan"],
-]
 
+def nl_to_sql(input: str, _test: bool = False):
+    db_path = "db/formula1.sqlite"
+    db = SQLDatabase.from_uri(f"sqlite:///{db_path}", sample_rows_in_table_info=3)
 
-# Main function
-def nl_to_sql(prompt: str):
-    db_path = "files/nba.sqlite"
-    db = SQLDatabase.from_uri(f"sqlite:///{db_path}", sample_rows_in_table_info=2)
-    llm = OpenAI(temperature=0, verbose=False)  # type: ignore
+    if not _test:
+        llm = ChatOpenAI(model="gpt-3.5-turbo", temperature=0, verbose=True)  # type: ignore
```

Comment:
This code could move to nl_to_sql_chatgpt function that calls nl_to_sql

Then move the stubs to the test function in the test file.
the test covers nl_to_sql function but not chatgpt function. That’s the nature of testing with stubs.

the reason is_test is not a best practice is that it mixes test code (stubs) inside production code, and adds complexity (a single if statement in this case)

WDYT?